agent.go

```diff
@@ -969,33 +965,13 @@ func makeUnaryInterceptor() grpc.UnaryServerInterceptor {
 				// without the "noise" of these calls.
 				span.SetTag("api-category", "interactive")
 			}
-		} else {
-			// Just log call details
-			message = req.(proto.Message)
-
-			agentLog.WithFields(logrus.Fields{
-				"request": grpcCall,
-				"req":     message.String()}).Debug("new request")
-			start = time.Now()
 		}
 
 		// Use the context which will provide the correct trace
 		// ordering, *NOT* the context provided to the function
 		// returned by this function.
 		resp, err = handler(getGRPCContext(), req)
 
-		if !tracing {
-			// Just log call details
-			elapsed = time.Since(start)
-			message = resp.(proto.Message)
-
-			logger := agentLog.WithFields(logrus.Fields{
-				"request":  info.FullMethod,
-				"duration": elapsed.String(),
-				"resp":     message.String()})
-			logger.Debug("request end")
```

Comment:
These are debug logs. Are you seeing it affecting CPUs in non-debug mode?

Reply:
@bergwolf I am seeing these logs in the debug mode only.  For the static policy, I was using, I saw a lot of these messages being printed to the journal, which made it quite difficult to see any logs I was interested in.

Reply:
Currently we have two logging scenarios:

- minimal logging (default)
- "firehose mode" / full logging (debug mode).

I think we need to adopt a more granular approach and allow multiple debug levels. We could do this by adding a new `agent.debug_level=[0-9]` (or maybe just `agent.debug_level={low,medium,high}` as we really shouldn't need 10 levels of debug :smile:?) These messages could then only be displayed for the "full" or "high" level of debug logging. All debug logs should then have an `debug-level=XXX` field added to allow admins to filter of course.

The question is what we set `agent.debug_level=` to if the agent is only passed an `agent.log=debug` option (probably "full"/"high" as that's what users would expect I think).